quickfixj-core/src/main/java/quickfix/DefaultSessionSchedule.java

```diff
@@ -41,10 +41,17 @@
     private final int[] weekdayOffsets;
     protected static final Logger LOG = LoggerFactory.getLogger(DefaultSessionSchedule.class);
 
+    //Cache recent time data to reduce creation of calendar objects
+    private ThreadLocal<Calendar> threadLocalCalendar;
```

Comment:
ThreadLocal because I'm fairly sure Calendar isn't thread-safe.
This keeps a calendar which we update with the new time we're checking against.